x-pack/platform/plugins/shared/lens/public/visualizations/heatmap/toolbar_component.tsx

```diff
@@ -173,8 +178,26 @@ export const HeatmapToolbar = memo(
                       },
                     });
                   }}
-                  isAxisLabelVisible={state?.gridConfig.isXAxisLabelVisible}
+                  isAxisLabelVisible={isXAxisLabelVisible}
                 />
+                {isXAxisLabelVisible && (
+                  <AxisLabelOrientationSelector
```

Comment:
Did you consider putting this option inside the `AxisTicksSettings` component? Meaning instead of creating a new `AxisLabelOrientationSelector` component you just add that logic to the existing `AxisTicksSettings` component.

https://github.com/elastic/kibana/blob/d96168c64f88547aa2f8aa1f991fbfda899218a0/x-pack/platform/plugins/shared/lens/public/shared_components/axis/ticks/axis_ticks_settings.tsx#L30-L62

I imagine these 2 component would always be used together. And the `AxisTicksSettings` component in name is already generalized or any axis tick setting such as orientation.

Reply:
That's a great improvement as a next step, but as this component is used by xy chart, let's not do it until https://github.com/elastic/kibana/issues/206040

Reply:
I also don't think we need to consider these options together right now because not every chart can support this feature (see goal/bullet graphs even if we are not considering hiding labels).
I can only suggest to change the `AxisTicksSettings` component name to be a bit more specific (like AxisTickVisibility) and in case, whenever we need to share this more broadly, we can have create a wrapper component that use both, but I also don't see much that need